### PR TITLE
Add Object derive macro

### PR DESCRIPTION
**Pull request by jplatte**
… that generates a free function for an object/interface type.

Part of  #1257.

### CODE REVIEW

File: uniffi_bindgen/src/interface/mod.rs [hunk 1]
```diff
@@ -614,6 +613,17 @@ impl Hash for ComponentInterface {
     }
 }
 
+fn get_or_insert_object<'a>(objects: &'a mut Vec<Object>, name: &str) -> &'a mut Object {
+    // The find-based way of writing this currently runs into a borrock error, so we use position
```
Comment (bendk) [inline uniffi_bindgen/src/interface/mod.rs:617]:
```suggestion
    // The find-based way of writing this currently runs into a borrow error, so we use position
```

**Review by bendk — APPROVED** [review]
Looks good to me, I just had a minor spelling suggestion.

File: uniffi_bindgen/src/interface/mod.rs (hunk 1 above)
Comment (jplatte) [reply uniffi_bindgen/src/interface/mod.rs:617]: It was supposed to be "borrowck", but I guess I should expand that to "borrow checker" 😄

**Comment by jplatte** [comment]
@bendk Can you re-review / merge? :)

**Comment by bendk** [comment]
No problem, the code looks great.

**Pull request merged**